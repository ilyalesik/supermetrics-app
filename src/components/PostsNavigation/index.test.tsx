import React from "react";
import renderer from "react-test-renderer";
import { PostsNavigation } from "./index";
import { fireEvent, render, screen } from "@testing-library/react";

it("renders PostsNavigation", () => {
  const tree = renderer
    .create(<PostsNavigation onAsc={console.log} onDesc={console.log} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("PostsNavigation onAsc/onDesc works", () => {
  const onAsc = jest.fn();
  const onDesc = jest.fn();
  render(<PostsNavigation onAsc={onAsc} onDesc={onDesc} />);
  fireEvent.click(screen.getByLabelText("asc-button"));
  fireEvent.click(screen.getByLabelText("desc-button"));
  expect(onAsc.mock.calls.length).toBe(1);
  expect(onDesc.mock.calls.length).toBe(1);
});
