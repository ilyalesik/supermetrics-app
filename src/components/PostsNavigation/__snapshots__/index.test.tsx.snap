// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PostsNavigation 1`] = `
<div
  className="posts-navigation"
>
  <button
    aria-label="asc-button"
    className="posts-navigation-button"
    onClick={[Function]}
  >
    asc
  </button>
  <button
    aria-label="desc-button"
    className="posts-navigation-button"
    onClick={[Function]}
  >
    desc
  </button>
</div>
`;
